### DISCUSSION TITLE
Testing created actions with remote download

### DISCUSSION THREAD

**Discussion by soumyamahunt** [community/community#64451] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi All! 

Currently I am creating action by following [this guide](https://docs.github.com/en/actions/creating-actions) and [this typescript action repo](https://github.com/actions/typescript-action). The sample action repo tests the action locally in [this workflow](https://github.com/actions/typescript-action/blob/main/.github/workflows/test.yml). I am wondering is there a way to test the action remotely, i.e the way action is actually going to be used:

```yml
- name: Execute
  uses: ${{ github. repository }}@${{ github.ref_name }}
```

Note that the above syntax doesn't work currently, I am getting:
> The workflow is not valid.

Is there any alternative way I can achieve this?

**Comment by nirav-gajera** [comment]
try uses-with:
```

- name: Execute
  uses-with:
    repo: ${{ github. repository }}
    ref: ${{ github.ref_name }}
```

**Reply by soumyamahunt** [reply]
Thanks @nirav-gajera, will try this. I couldn't find any documentation regarding this attribute in official syntax docs. Do you have any resources that can give all the supported options for this attribute?

**Reply by soumyamahunt** [reply]
@nirav-gajera when trying this getting the following error:
> every step must define a `uses` or `run` key

**Reply by nirav-gajera** [reply]
You can fix this by adding either a `uses` key or a `run` key to the step.

For ex use the `actions/checkout@v2` action to checkout the repository:

```
- name: Checkout repository
  uses: actions/checkout@v2
```

The following step would run the `npm install` command:

```
- name: Install dependencies
  run: npm install
```

Once you have added either a `uses` key or a `run` key to the step, the error should be resolved.

**Reply by nirav-gajera** [reply]
ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses
https://stackoverflow.com/questions/70200476/github-actions-every-step-must-define-a-uses-or-run-key

**Reply by Rockyread** [reply]
discussions

**Reply by soumyamahunt** [reply]
> You can fix this by adding either a `uses` key or a `run` key to the step.
> 
> For ex use the `actions/checkout@v2` action to checkout the repository:
> 
> ```
> - name: Checkout repository
>   uses: actions/checkout@v2
> ```
> 
> The following step would run the `npm install` command:
> 
> ```
> - name: Install dependencies
>   run: npm install
> ```
> 
> Once you have added either a `uses` key or a `run` key to the step, the error should be resolved.

So the snippet you provided should be modified this way:
```yml
- name: Execute
  uses: ${{ github. repository }}@${{ github.ref_name }}
  uses-with:
    repo: ${{ github. repository }}
    ref: ${{ github.ref_name }}
```

@nirav-gajera Can you tell me if this is correct or any further changes I need to make?

> ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses https://stackoverflow.com/questions/70200476/github-actions-every-step-must-define-a-uses-or-run-key

I couldn't find anything related to `uses-with:` in this doc, can you provide something related to `uses-with:` as well?

**Reply by nirav-gajera** [reply]
Sorry bro, I only have this level of knowledge. You might want to try posting your query on Stack Overflow, or you can wait for someone to respond on this

**Comment by soumyamahunt** [comment, marked as answer]
I have resolved this by wrapping the dynamic action inside a composite action present at a static path, this composite action is populated with dynamic content before running the action. Full implementation can be found [here](https://github.com/SwiftyLab/setup-swift/blob/86e007cef8044fb2f809e74b67071f5c3864a1e4/.github/workflows/main.yml#L240-L305).
